tests/testdata/provider/testdata_pg_bigint_pk.sql

```diff
@@ -119,13 +119,58 @@ INSERT INTO qgis_test.tb_test_compound_pk (pk1, pk2, value, geom) VALUES
     (2, 1, 'test 3', ST_SetSRID(ST_Point(-47.902, -15.763), 4326)),
     (2, 2, 'test 4', ST_SetSRID(ST_Point(-47.952, -15.781), 4326));
 
+CREATE TABLE qgis_test.tb_test_composite_float_pk
+(
+    pk1 INTEGER,
+    pk2 BIGINT,
+    pk3 REAL,
+    value VARCHAR(16),
+    geom geometry(Point, 4326),
+    PRIMARY KEY (pk1, pk2, pk3)
+);
+
+INSERT INTO qgis_test.tb_test_composite_float_pk (pk1, pk2, pk3, value, geom) VALUES
+    (1, 1, 1.0,         'test 1', ST_SetSRID(ST_Point(-47.930, -15.818), 4326)),
+    (1, 2, 3.141592741, 'test 2', ST_SetSRID(ST_Point(-47.887, -15.864), 4326)),
+    (2, 2, 2.718281828, 'test 3', ST_SetSRID(ST_Point(-47.902, -15.763), 4326)),
+    (2, 2, 1.0,         'test 4', ST_SetSRID(ST_Point(-47.952, -15.781), 4326));
+
 CREATE TABLE qgis_test.tb_test_float_pk
 (
     pk REAL PRIMARY KEY,
     value VARCHAR(16),
     geom geometry(Point, 4326)
 );
-
+-- those values (pi, Euler's, and a third) will be truncated/rounded to fit
+-- PostgreSQL's internal type size. REAL is IEEE-754 4 bytes (32 bit).
 INSERT INTO qgis_test.tb_test_float_pk (pk, value, geom) VALUES
-    (3.14159, 'first test',  ST_SetSRID(ST_Point(-47.887, -15.864), 4326)),
-    (2.71828, 'second test', ST_SetSRID(ST_Point(-47.902, -15.763), 4326));
+    (3.141592653589793238462643383279502884197169399375105820974944592307816406286, 'first teste', ST_SetSRID(ST_Point(-47.887, -15.864), 4326)),
+    (2.718281828459045235360287471352662497757247093699959574966967627724076630353, 'second test', ST_SetSRID(ST_Point(-47.902, -15.763), 4326)),
+    (1.333333333333333333333333333333333333333333333333333333333333333333333333333, 'third teste', ST_SetSRID(ST_Point(-47.751, -15.644), 4326));
+
+CREATE TABLE qgis_test.tb_test_double_pk
+(
+    pk DOUBLE PRECISION PRIMARY KEY,
+    value VARCHAR(16),
+    geom geometry(Point, 4326)
+);
+-- those values (pi, Euler's, and a third) will be truncated/rounded to fit
+-- PostgreSQL's internal type size. DOUBLE PRECISION is IEEE-754 8 bytes (64 bit).
+INSERT INTO qgis_test.tb_test_double_pk (pk, value, geom) VALUES
+    (3.141592653589793238462643383279502884197169399375105820974944592307816406286, 'first teste', ST_SetSRID(ST_Point(-47.887, -15.864), 4326)),
+    (2.718281828459045235360287471352662497757247093699959574966967627724076630353, 'second test', ST_SetSRID(ST_Point(-47.902, -15.763), 4326)),
+    (1.333333333333333333333333333333333333333333333333333333333333333333333333333, 'third teste', ST_SetSRID(ST_Point(-47.751, -15.644), 4326));
+
+CREATE TABLE qgis_test.tb_test_numeric_pk
+(
+    pk NUMERIC(41, 40) PRIMARY KEY,
+    value VARCHAR(16),
+    geom geometry(Point, 4326)
+);
+-- those values (pi, Euler's, and a third) will be truncated to fit
```

Comment:
is this still checked?

Reply:
It is not, I removed the check for NUMERIC in the Python code, but forgot to delete the testdata. c682440 removes it, thanks!